### ISSUE TITLE
Undo in conversations in personal settings is not valid for ios and Android

### ISSUE THREAD

**Issue by HandsomeBoby** [nextcloud/server#8521]
### Steps to reproduce
1.Click settings
2.Click on personal Sessions
3.Clicking the undo button for Android or ios conversation is not valid 

### Expected behaviour
Tell us what should happen
Android and ios can cancel the login operation
### Actual behaviour
Tell us what happens instead
Can not cancel
### Server configuration

**Operating system**:

**Web server:**

**Database:**

**PHP version:**

**Nextcloud version:** (see Nextcloud admin page)

**Updated from an older Nextcloud/ownCloud or fresh install:**

**Where did you install Nextcloud from:**

**Signing status:**
<details>
<summary>Signing status</summary>

```
Login as admin user into your Nextcloud and access 
http://example.com/index.php/settings/integrity/failed 
paste the results here.
```
</details>

**List of activated apps:**
<details>
<summary>App list</summary>

```
If you have access to your command line run e.g.:
sudo -u www-data php occ app:list
from within your Nextcloud installation folder
```
</details>

**Nextcloud configuration:**
<details>
<summary>Config report</summary>

```
If you have access to your command line run e.g.:
sudo -u www-data php occ config:list system
from within your Nextcloud installation folder

or 

Insert your config.php content here. 
Make sure to remove all sensitive content such as passwords. (e.g. database password, passwordsalt, secret, smtp password, …)
```
</details>

**Are you using external storage, if yes which one:** local/smb/sftp/...

**Are you using encryption:** yes/no

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...

#### LDAP configuration (delete this part if not used)
<details>
<summary>LDAP config</summary>

```
With access to your command line run e.g.:
sudo -u www-data php occ ldap:show-config
from within your Nextcloud installation folder

Without access to your command line download the data/owncloud.db to your local
computer or access your SQL server remotely and run the select query:
SELECT * FROM `oc_appconfig` WHERE `appid` = 'user_ldap';


Eventually replace sensitive data as the name/IP-address of your LDAP server or groups.
```
</details>

### Client configuration
**Browser:**

**Operating system:**

#### Web server error log
<details>
<summary>Web server error log</summary>

```
Insert your webserver log here
```
</details>

#### Nextcloud log (data/nextcloud.log)
<details>
<summary>Nextcloud log</summary>

```
Insert your Nextcloud log here
```
</details>

#### Browser log
<details>
<summary>Browser log</summary>

```
Insert your browser log here, this could for example include:

a) The javascript console log
b) The network log
c) ...
```
</details>

**Comment by HandsomeBoby** [comment]
@blizzz @MorrisJobke  Thanks!

**Comment by MorrisJobke** [comment]
> 3.Clicking the undo button for Android or ios conversation is not valid

I don't know what you mean with this?

Do you mean this:

![bildschirmfoto 2018-02-26 um 09 06 29](https://user-images.githubusercontent.com/245432/36659318-66a21d42-1ad4-11e8-9e2f-ffdb1bf8df7d.png)

And "does not work" is not the best description. Could you be a bit more precise and explain what happens?

**Comment by HandsomeBoby** [comment]
@MorrisJobke 
Yes, I think this is a bug. About the web the revoke(disconnected) is works, but the client including android, ios, windows and mac the revoke button is doesn't work. I have no ideas to solve this problem.

**Comment by MorrisJobke** [comment]
> Yes, I think this is a bug. About the web the revoke(disconnected) is works, but the client including android, ios, windows and mac the revoke button is doesn't work. Log out of the web after the disconnect, but the client can not. I have no ideas to solve this problem.

I don't understand you properly. What is a bug here? What do you do exactly and what does not work?

**Comment by supermanlinux** [comment]
@MorrisJobke I have the same question.First you login the nextcloud server using ios or android client,when you click the revoke button,the ios or android client can use normaly,not goto the login page.But when you login the nextcloud server using browser like firefox or google chrome,then you click the revoke button,then you browser jump to login page,you must login again then you can use nextcloud.

**Comment by MorrisJobke** [comment]
> @MorrisJobke I have the same question.First you login the nextcloud server using ios or android client,when you click the revoke button,the ios or android client can use normaly,not goto the login page.But when you login the nextcloud server using browser like firefox or google chrome,then you click the revoke button,then you browser jump to login page,you must login again then you can use nextcloud.

This is because we currently work on bring all apps to use app passwords. The newest Android and iOS apps already do this, but also support the old way of storing the username and password and not username and app password. Thus revoking an "app with the old way of authentication" will result in just a re-auth, because the actual password was stored and it just fetches a new session. With the new approach this no longer would be the case and the apps would be signed out.

Android supports it from version 3.0 and onwards. On iOS I'm not 100% sure it is already available in the appstore, but with 2.20.3 (from Testflight) it works.

**Comment by MorrisJobke** [comment]
Okay - then this is solved: you need to re-authenticate using a new client to be able to properly revoke the login details. For older client (or newer clients with the old way of authentication) this will not work.
